sdk/identity/identity/Troubleshooting.md

```diff
@@ -1,48 +1,85 @@
 
-If you're using app registration to authenticate the service, ensure the app registration has the correct permissions and role assignments in the service you want to use. For example, if you want to have access to the Azure App Configuration service through Azure Active Directory (Azure AD), make sure your app registration has the permissions and the role assignments for access to Azure AD. You can either be assigned the role directly or be in a group that is assigned the role. The "Contributor" and the "Owner" roles allow you to manage the App Configuration resource. In this case, you can either use "App Configuration Data Owner" directly on the user or the Azure AD group. Alternatively, use "Owner" on the Azure AD group. While the App Configuration data can be accessed using access keys, these keys don't grant direct access to the data using Azure AD.
+### AuthenticationRequiredError
 
-## Troubleshoot default Azure credential authentication issues
+Errors arising from authentication issues can be thrown on any service client method that makes a request to the service. This is because the token is requested from the credential on the first call to the service and on any subsequent call that need to refresh the token.
 
-### Credential unavailable
+To distinguish these failures from failures in the service client, Azure Identity classes raise the `AuthenticationRequiredError` with details describing the source of the error in the exception message and possibly the error message. Depending on the application, these errors may or may not be recoverable.
 
-The `DefaultAzureCredential` attempts to retrieve an access token by sequentially invoking a chain of credentials. In this scenario, the `CredentialUnavailableError` signifies that all credentials in the chain failed to retrieve the token in the current environment setup/configuration. You need to follow the configuration instructions for the respective credential you're looking to use via the `DefaultAzureCredential` chain, so that the credential can work in your environment.
+```ts
+import @azure/identity
+import @azure/keyvault-secrets
+
+// Create a key client using the DefaultAzureCredential
+ const keyVaultUrl = `https://key-vault-name.vault.azure.net`;
+ const client = new KeyClient(keyVaultUrl, new DefaultAzureCredential());
+
+ try {
+// Retrieving the properties of the existing keys in that specific Key Vault.
+    console.log(await client.listPropertiesOfKeys().next());
+ } catch (error) {
+    console.log("Authentication Failed", error.message);
+ }
+```
+
+### CredentialUnavailableError
+
+The `CredentialUnavailableError` is used to indicate that the credential can’t authenticate in the current environment, due to lack of required configuration or setup. This error is also used as a signal to chained credential types, such as `DefaultAzureCredential` and `ChainedTokenCredential`, that the chained credential should continue to try other credential types later in the chain.
```

Comment:
I believe we should put emphasis on the environment here, since a credential may be unavailable if the credential can’t reach to the authentication endpoint.

Reply:
@sadasant Yeah we have mentioned that. "current environment". Do you have any suggestions like making it bold or changing the wording a bit to make it obvious?